### PR TITLE
Type command (#441)

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @AlexanderMoskovkin @AndreyBelym @VasilyStrelyaev

### CODE REVIEW

File: src/errors/test-run/templates.js [hunk 1]
```diff
@@ -78,6 +78,12 @@ export default {
         ${err.getCallsiteMarkup()}
     `),
 
+    [TYPE.actionTextTypeError]: err => markup(err, `
+        Action text is expected to be a string, but it was <code>${err.actualType}</code>.
```
Comment (VasilyStrelyaev) [inline src/errors/test-run/templates.js]:
```
        The text to type is expected to be a string, but it was <code>${err.actualType}</code>.
```

Comment (VasilyStrelyaev) [inline src/errors/test-run/templates.js]: Got better or worse, @inikulin @AlexanderMoskovkin ?

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

File: src/errors/test-run/templates.js (hunk 1 above)
Comment (AlexanderMoskovkin) [inline src/errors/test-run/templates.js]: I like the new text more

File: src/errors/test-run/index.js [hunk 2]
```diff
@@ -118,6 +118,14 @@ export class ActionOptionsTypeError extends ActionError {
     }
 }
 
+export class ActionTextTypeError extends ActionError {
```
Comment (AlexanderMoskovkin) [inline src/errors/test-run/index.js]: I suppose this error is specific for the `type` action so maybe we should use it in the error name? `TypeTextArgumentTypeError`, `TypeTextTypeError` or something better.. /cc @inikulin @VasilyStrelyaev

File: src/test-run/commands/index.js [hunk 3]
```diff
@@ -45,6 +50,15 @@ function integer (option, val) {
         throw new ActionIntegerOptionError(option, val);
 }
 
+function text (option, val) {
+    var type = typeof val;
+
+    if (type !== 'string')
+        throw new ActionTextTypeError(type);
+    else if (!val.length)
```
Comment (AlexanderMoskovkin) [inline src/test-run/commands/index.js]: unnecessary `else`

File: src/errors/test-run/index.js (hunk 2 above)
Comment (inikulin) [inline src/errors/test-run/index.js]: `TypeActionTextArgumentTypeError`

File: src/errors/test-run/templates.js (hunk 1 above)
Comment (inikulin) [inline src/errors/test-run/templates.js]: Better

File: src/errors/test-run/index.js (hunk 2 above)
Comment (AlexanderMoskovkin) [inline src/errors/test-run/index.js]: It will be `ActionStringArgumentError` (discussed in private)

**Comment by AlexanderMoskovkin** [comment]
FYI: rebase required

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

File: src/test-run/commands/index.js [hunk 4]
```diff
@@ -45,6 +51,16 @@ function integer (option, val) {
         throw new ActionIntegerOptionError(option, val);
 }
 
+function text (option, val) {
```
Comment (AlexanderMoskovkin) [inline src/test-run/commands/index.js]: it should not be `text`. It's something like `nonEmptyArgumentString`

Comment (inikulin) [inline src/test-run/commands/index.js]: `nonEmptyStringArgument` =)

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
lgtm

**Comment by helen-dikareva** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm
/cc @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**